Automattic/wp-calypso
# composite-checkout: Log error boundary events to analytics

#### Changes proposed in this Pull Request

When an unexpected error is thrown inside composite checkout components, it triggers one of various layers of [React Error Boundaries](https://reactjs.org/docs/error-boundaries.html) to protect the page.

This PR adds Tracks events for those errors so we can see when they occur.

#### Testing instructions

Since there are (hopefully!) no fatal errors currently lurking in the code, you'll have to edit the code to throw one in order to test this properly.

- Add `throw new Error('testing error');` into a component such as `WPCheckout`.
- Load calypso and enable analytics debugging by typing the following in your browser console `localStorage.setItem('debug', 'calypso:analytics')`. Then reload the page.
- Visit composite checkout. You should see "Sorry, there was an error loading this page".
- Verify that you see the `calypso_checkout_composite_page_load_error` event in the debug output in the console.
